packages/vendure-plugin-store-credit/src/services/wallet.service.ts

```
@@ -74,17 +75,35 @@ export class WalletService {
       Wallet,
       wallet.id,
       unique([defaultChannel.id, ctx.channel.id])
     );
-    return assertFound(this.findOne(ctx, savedWallet.id));
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const foundWallet = await this.findOne(ctx, savedWallet!.id);
+    if (!foundWallet) {
+      throw new InternalServerError('Wallet was not found after creation');
+    }
+    return foundWallet;
+    return assertFound(this.findOne(ctx, savedWallet!.id));
```

Biome (2.3.13)

[error] This code will never be reached ...

... because this statement will return from the function beforehand


(lint/correctness/noUnreachable)